great_expectations/execution_engine/sqlalchemy_batch_data.py

```diff
@@ -106,11 +107,12 @@ def __init__(
             raise ValueError(
                 "schema_name can only be used with table_name. Use temp_table_schema_name to provide a target schema for creating a temporary table."
             )
+        dialect: GESqlDialect = GESqlDialect(engine.dialect.name.lower())
         if table_name:
             # Suggestion: pull this block out as its own _function
             if use_quoted_name:
                 table_name = quoted_name(table_name, quote=True)
-            if engine.dialect.name.lower() == "bigquery":
+            if dialect == GESqlDialect.BIGQUERY:
```

Comment:
Just a thought - what we made the `__eq__` on the enum deal with sanitizing the string so we could have case-insensitive comparisons?

Reply:
Oh interesting - well in this case I'm planning to remove all of the strings in a future PR so that we are only using the enum. So once the enum is created, it'll only be compared with instances of the enum.